FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -84,20 +84,50 @@ public static void ValidateConfiguration(this IServiceCollection services)
                 throw new ConfigurationErrorException($"Target azure container name can not be empty.");
             }
 
-            FilterConfiguration filterConfiguration;
+            FilterLocation filterLocation;
             try
             {
-                filterConfiguration = services
+                filterLocation = services
                     .BuildServiceProvider()
-                    .GetRequiredService<IOptions<FilterConfiguration>>()
+                    .GetRequiredService<IOptions<FilterLocation>>()
                     .Value;
             }
             catch (Exception ex)
             {
-                throw new ConfigurationErrorException("Failed to parse filter configuration", ex);
+                throw new ConfigurationErrorException("Failed to parse filter location", ex);
             }
 
-            ValidateFilterConfiguration(filterConfiguration);
+            if (filterLocation.EnableExternalFilter)
```

Comment:
check null?

Reply:
A bool variable will never be null?

Reply:
filterLocation is possible none?

Reply:
Will not be null, it will return default if not registration.